build.py

```diff
@@ -2362,6 +2343,12 @@ def enable_all():
         required=False,
         help="Override specified backend CMake argument in the build as <backend>:<name>=<value>. The argument is passed to CMake as -D<name>=<value>. This flag only impacts CMake arguments that are used by build.py. To unconditionally add a CMake argument to the backend build use --extra-backend-cmake-arg.",
     )
+    parser.add_argument(
```

Comment:
@mc-nv Do we have a similar practice of setting library versions through`build.py` args?

Reply:
yes, we can keep adding. This is also meant to serve as an good example to do that going forward

Reply:
What do you mean under "library versions"?

Reply:
This `add_argument` adds `vllm-version`, i.e. with this PR we would be able to specify vllm library version through `build.py` argument. IIRC, we didn't allow anything like this, but rather had a data structure with specified versions

Reply:
I have no answer to it, I would like to have a consistency and if we give control under one value trough the argument parser then make sense to extend functionality for any other argument value in the data structure.  

By giving version control to `build.py` with argument, we created precedent: 
- any issue with build, must include check between version in `build.py` and with argument `vllm-version` given to it during the execution.

I have no guidance on it, I don't like wrappers at all and would  rather get rid of it. 
Current scenario brings kind of loop back from `vllm_backend` to `server`. 

From build stand point as long as `vllm_backend` has nothing to include from server, it should be built separately.

Reply:
I would like to have a consistency as well, thus I recommend to either delete this new flag, or handle all other libraries

Reply:
I've added arg parser for all libraries

Reply:
can we additionally mention the usage for them in the docs?

Reply:
can be in a follow-up pr